sample-implementation.js

```diff
@@ -1,120 +0,0 @@
-/**
- * DISCLAIMER
- *
- * This implementation does not have any safety checks or optimizations.
- * Instead, the code is highly readable.
- */
-
-import _ from 'lodash';
-
-export function isComposable(obj) {
-  return _.isFunction(obj) && _.isFunction(obj.compose);
-}
-export function isDescriptor(obj) {
-  return _.isObject(obj);
-}
-function mixDescriptors(...args) {
```

Comment:
I think my unfinished revision just about cuts this in half using Object.assign and mapping properties in advance. Probably more performant, too?